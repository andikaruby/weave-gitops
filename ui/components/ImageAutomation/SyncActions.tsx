import React from "react";
import { useSyncFluxObject } from "../../hooks/automations";
import { useToggleSuspend } from "../../hooks/flux";
import { Kind } from "../../lib/api/core/types.pb";
import Button from "../Button";
import Flex from "../Flex";
import Spacer from "../Spacer";
import SyncButton from "../SyncButton";

interface Props {
  name?: string;
  namespace?: string;
  clusterName?: string;
  kind?: Kind;
  suspended?: boolean;
  wide?: boolean;
  hideDropdown?: boolean;
}

const SyncActions = ({
  name,
  namespace,
  clusterName,
  kind,
  suspended,
  wide,
  hideDropdown,
}: Props) => {
  const suspend = useToggleSuspend(
    {
      objects: [
        {
          name,
          namespace,
          clusterName,
          kind: kind,
        },
      ],
      suspend: !suspended,
    },
    "object"
  );

  const sync = useSyncFluxObject([
    {
      name,
      namespace,
      clusterName,
      kind: kind,
    },
  ]);

  const syncHandler = hideDropdown
    ? () => sync.mutateAsync({ withSource: false })
    : (opts) => sync.mutateAsync(opts);

  return (
    <Flex wide={wide} start>
      <SyncButton
        onClick={syncHandler}
        loading={sync.isLoading}
        disabled={suspended}
        hideDropdown={hideDropdown}
      />
      <Spacer padding="xs" />
      <Button onClick={() => suspend.mutateAsync()} loading={suspend.isLoading}>
        {suspended ? "Resume" : "Suspend"}
      </Button>
    </Flex>
  );
};

export default SyncActions;
