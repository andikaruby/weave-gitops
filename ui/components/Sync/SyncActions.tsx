import React from "react";
import styled from "styled-components";
import { useSyncFluxObject } from "../../hooks/automations";
import { useToggleSuspend } from "../../hooks/flux";
import { Kind } from "../../lib/api/core/types.pb";
import SyncControls, { SyncType } from "./SyncControls";

interface Props {
  name?: string;
  namespace?: string;
  clusterName?: string;
  kind?: Kind;
  suspended?: boolean;
  hideSyncOptions?: boolean;
  customActions?: JSX.Element[];
  className?: string;
}

const SyncActions = ({
  name,
  namespace,
  clusterName,
  kind,
  suspended,
  hideSyncOptions,
  customActions,
  className,
}: Props) => {
  const sync = useSyncFluxObject([
    {
      name,
      namespace,
      clusterName,
      kind: kind,
    },
  ]);

  const syncHandler = (syncType: SyncType) => {
    sync.mutateAsync({ withSource: syncType === SyncType.WithSource });
  };

  const objects = [
    {
      name,
      namespace,
      clusterName,
      kind: kind,
    },
  ];

  const suspend = useToggleSuspend(
    {
      objects: objects,
      suspend: true,
    },
    "object"
  );

  const resume = useToggleSuspend(
    {
      objects: objects,
      suspend: false,
    },
    "object"
  );

  return (
    <SyncControls
      className={className}
      hideSyncOptions={hideSyncOptions}
      syncLoading={sync.isLoading}
      syncDisabled={suspended}
      suspendDisabled={suspend.isLoading || suspended}
      resumeDisabled={resume.isLoading || !suspended}
      customActions={customActions}
      onSyncClick={syncHandler}
      onSuspendClick={() => suspend.mutateAsync()}
      onResumeClick={() => resume.mutateAsync()}
    />
  );
};

export default styled(SyncActions)`
  width: 50%;
  min-width: fit-content;
`;
