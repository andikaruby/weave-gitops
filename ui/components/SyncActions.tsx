import React from "react";
import { useSyncFluxObject } from "../hooks/automations";
import { useToggleSuspend } from "../hooks/flux";
import { Kind } from "../lib/api/core/types.pb";
import Button from "./Button";
import CustomActions from "./CustomActions";
import Flex from "./Flex";
import Spacer from "./Spacer";
import SyncButton from "./SyncButton";

interface Props {
  name?: string;
  namespace?: string;
  clusterName?: string;
  kind?: Kind;
  suspended?: boolean;
  wide?: boolean;
  hideDropdown?: boolean;
  customActions?: JSX.Element[];
}

const SyncActions = ({
  name,
  namespace,
  clusterName,
  kind,
  suspended,
  wide,
  hideDropdown,
  customActions,
}: Props) => {
  const suspend = useToggleSuspend(
    {
      objects: [
        {
          name,
          namespace,
          clusterName,
          kind: kind,
        },
      ],
      suspend: !suspended,
    },
    "object"
  );

  const sync = useSyncFluxObject([
    {
      name,
      namespace,
      clusterName,
      kind: kind,
    },
  ]);

  const syncHandler = hideDropdown
    ? () => sync.mutateAsync({ withSource: false })
    : (opts) => sync.mutateAsync(opts);

  return (
    <Flex wide={wide} start>
      <SyncButton
        onClick={syncHandler}
        loading={sync.isLoading}
        disabled={suspended}
        hideDropdown={hideDropdown}
      />
      <Spacer padding="xs" />
      <Button onClick={() => suspend.mutateAsync()} loading={suspend.isLoading}>
        {suspended ? "Resume" : "Suspend"}
      </Button>
      <CustomActions actions={customActions} />
    </Flex>
  );
};

export default SyncActions;
