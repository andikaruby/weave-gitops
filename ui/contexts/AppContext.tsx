import * as React from "react";
import { useHistory } from "react-router-dom";
import { formatURL } from "../lib/nav";
import { PageRoute, V2Routes } from "../lib/types";
import { notifySuccess } from "../lib/utils";

type AppState = {
  error: null | { fatal: boolean; message: string; detail?: string };
};

type AppSettings = {
  renderFooter: boolean;
};

export type LinkResolver = (incoming: string) => string;

export function defaultLinkResolver(incoming: string): string {
  return incoming;
}

export type AppContextType = {
  userConfigRepoName: string;
  doAsyncError: (message: string, detail: string) => void;
  clearAsyncError: () => void;
  appState: AppState;
  settings: AppSettings;
  linkResolver: LinkResolver;
  navigate: {
    internal: (page: PageRoute | V2Routes, query?: any) => void;
    external: (url: string) => void;
  };
  notifySuccess: typeof notifySuccess;
  request: typeof window.fetch;
};

export const AppContext = React.createContext<AppContextType>(
  null as AppContextType
);

export interface AppProps {
  linkResolver?: LinkResolver;
  children?: any;
  renderFooter?: boolean;
  notifySuccess?: typeof notifySuccess;
}

export default function AppContextProvider({ ...props }: AppProps) {
  const history = useHistory();
  const [appState, setAppState] = React.useState({
    error: null,
  });

  const clearAsyncError = () => {
    setAppState({
      ...appState,
      error: null,
    });
  };

  React.useEffect(() => {
    // clear the error state on navigation
    clearAsyncError();
  }, [window.location]);

  const doAsyncError = (message: string, detail: string) => {
    console.error(message);
    setAppState({
      ...appState,
      error: { message, detail },
    });
  };

  const value: AppContextType = {
    userConfigRepoName: "wego-github-jlw-config-repo",
    doAsyncError,
    clearAsyncError,
    appState,
    linkResolver: props.linkResolver || defaultLinkResolver,
    notifySuccess: props.notifySuccess || notifySuccess,
    settings: {
      renderFooter: props.renderFooter,
    },
    navigate: {
      internal: (page: PageRoute, query?: any) => {
        const u = formatURL(page, query);

        history.push(u);
      },
      external: (url) => {
        if (process.env.NODE_ENV === "test") {
          return;
        }
        window.location.href = url;
      },
    },
    request: window.fetch,
  };

  return <AppContext.Provider {...props} value={value} />;
}
