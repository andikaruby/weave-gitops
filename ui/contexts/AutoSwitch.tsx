import * as React from "react";
import { Redirect, Route, Switch } from "react-router-dom";
import SignIn from "../pages/SignIn";

const AUTH_PATH_SIGNIN = "/sign_in";

export const AuthSwitch: React.FC = () => {
  const handle404 = () => <Redirect exact={true} to={AUTH_PATH_SIGNIN} />;

  console.log("In the Auth Switch");

  return (
    <Switch>
      <Route component={SignIn} exact={true} path={AUTH_PATH_SIGNIN} />
      <Route render={handle404} />
    </Switch>
  );
};
