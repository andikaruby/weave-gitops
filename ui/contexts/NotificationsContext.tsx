import { createContext, Dispatch, useContext } from "react";

export interface NotificationData {
  message: { text?: string; component?: JSX.Element };
  severity?: "success" | "error" | "warning" | "info";
  display?: "top" | "bottom";
}

type NotificationContext = {
  notifications: NotificationData[] | [];
  setNotifications: Dispatch<React.SetStateAction<NotificationData[] | []>>;
};

export const Notification = createContext<NotificationContext | null>(null);

export function useNotifications() {
  return useContext(Notification) as NotificationContext;
}
