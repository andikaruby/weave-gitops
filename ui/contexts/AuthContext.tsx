import * as React from "react";
import LoadingPage from "../components/LoadingPage";
import { AuthSwitch } from "./AutoSwitch";

const USER_INFO = "/oauth2/userinfo";
const SIGN_IN = "/oauth2/sign_in";

export type AuthContext = {
  signIn: (data: any) => void;
  userInfo: {
    email: string;
    groups: string[];
  };
};

export const Auth = React.createContext<AuthContext | null>(null);

export default function AuthContextProvider({ children }) {
  const [userInfo, setUserInfo] = React.useState<{
    email: string;
    groups: string[];
  } | null>(null);
  const [loading, setLoading] = React.useState<boolean>(false);

  const signIn = React.useCallback((data) => {
    console.log(data);
    fetch(SIGN_IN, {
      method: "POST",
      body: JSON.stringify(data),
    })
      .then((response) => {
        console.log(response);
        return response.json();
      })
      .then((data) => {
        console.log(data);
        //redirect to "/""
      })
      .catch((err) => console.log(err));
  }, []);

  const getUserInfo = React.useCallback(() => {
    // setLoading(true);
    fetch(USER_INFO)
      .then((response) => {
        return response.json();
      })
      .then((data) => setUserInfo({ email: data.email, groups: [] }))
      .catch((err) => {
        console.log(err);
        if (err.code === "401") {
          setUserInfo(null);
        }
      });
    // .finally(() => setLoading(false));
  }, []);

  console.log(userInfo?.email);

  React.useEffect(() => {
    getUserInfo();
  }, [getUserInfo]);

  return (
    <Auth.Provider value={{ signIn, userInfo }}>
      {/* {loading ? <LoadingPage /> : null} */}
      {userInfo?.email && !loading ? children : <AuthSwitch />}
    </Auth.Provider>
  );
}
