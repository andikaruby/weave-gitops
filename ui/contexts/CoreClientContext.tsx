import qs from "query-string";
import * as React from "react";
import { useQuery } from "react-query";
import { Core, GetFeatureFlagsResponse } from "../lib/api/core/core.pb";
import { RequestError } from "../lib/types";
import { TokenRefreshWrapper } from "../lib/requests";
import { AuthRoutes } from "./AuthContext";

type Props = {
  api: typeof Core;
  children: any;
};

export type FeatureFlags = { [key: string]: string };

export type CoreClientContextType = {
  api: typeof Core;
  featureFlags: FeatureFlags;
};

export const CoreClientContext =
  React.createContext<CoreClientContextType | null>(null);

function FeatureFlags(api) {
  const { data } = useQuery<GetFeatureFlagsResponse, RequestError>(
    "feature_flags",
    () => api.GetFeatureFlags({}),
    {
      staleTime: Infinity,
      cacheTime: Infinity,
    }
  );
  return data?.flags;
}

export async function refreshToken() {
  const res = await fetch("/oauth2/refresh", { method: "POST" });
  if (!res.ok) {
    window.location.replace(
      AuthRoutes.AUTH_PATH_SIGNIN +
        "?" +
        qs.stringify({
          redirect: location.pathname + location.search,
        })
    );
  }
}

export function UnAuthorizedInterceptor(api: typeof Core): typeof Core {
  return TokenRefreshWrapper.wrap(api, refreshToken);
}

export default function CoreClientContextProvider({ api, children }: Props) {
  const wrapped = UnAuthorizedInterceptor(api);

  return (
    <CoreClientContext.Provider
      value={{ api: wrapped, featureFlags: FeatureFlags(wrapped) }}
    >
      {children}
    </CoreClientContext.Provider>
  );
}
