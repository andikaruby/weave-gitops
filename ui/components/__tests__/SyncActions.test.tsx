import "jest-styled-components";
import React from "react";
import renderer from "react-test-renderer";
import {
  createCoreMockClient,
  withContext,
  withTheme,
} from "../../lib/test-utils";
import SyncActions from "../SyncActions";
import { CoreClientContext } from "../../contexts/CoreClientContext";

describe("SyncActions", () => {
  describe("snapshots", () => {
    const mockContext = { api: createCoreMockClient({}), featureFlags: {} };

    it("non-suspended", () => {
      const tree = renderer
        .create(
          withTheme(
            withContext(
              <CoreClientContext.Provider value={mockContext}>
                <SyncActions />
              </CoreClientContext.Provider>,
              "/",
              {}
            )
          )
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("suspended", () => {
      const tree = renderer
        .create(
          withTheme(
            withContext(
              <CoreClientContext.Provider value={mockContext}>
                <SyncActions suspended />
              </CoreClientContext.Provider>,
              "/",
              {}
            )
          )
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("wide", () => {
      const tree = renderer
        .create(
          withTheme(
            withContext(
              <CoreClientContext.Provider value={mockContext}>
                <SyncActions wide />
              </CoreClientContext.Provider>,
              "/",
              {}
            )
          )
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("hideDropdown", () => {
      const tree = renderer
        .create(
          withTheme(
            withContext(
              <CoreClientContext.Provider value={mockContext}>
                <SyncActions hideDropdown />
              </CoreClientContext.Provider>,
              "/",
              {}
            )
          )
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
