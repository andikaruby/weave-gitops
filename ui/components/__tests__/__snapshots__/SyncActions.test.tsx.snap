// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SyncActions snapshots hideDropdown 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.c6 {
  padding: 8px;
}

.c3.MuiButton-root {
  height: 32px;
  font-size: 12px;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  line-height: 1;
  border-radius: 2px;
  font-weight: 600;
}

.c3.MuiButton-outlined {
  padding: 8px 12px;
}

.c4.MuiButton-outlined {
  margin-right: 0;
}

.c4.MuiButton-outlined.Mui-disabled {
  border-right: none;
}

.c5 {
  position: absolute;
  overflow: hidden;
  background: #fff;
  height: 0px;
  -webkit-transition-property: height;
  transition-property: height;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  z-index: 1;
}

.c1 {
  width: 50%;
}

<div
  className="c0 c1"
>
  <div
    className="SyncButton"
    style={
      Object {
        "display": "inline-block",
        "position": "relative",
      }
    }
  >
    <div
      className="c2"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined c3 c4 MuiButton-outlinedPrimary MuiButton-disableElevation"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Sync
        </span>
      </button>
    </div>
    <div
      className="c2 c5"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined c3 MuiButton-outlinedPrimary MuiButton-disableElevation"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "whiteSpace": "nowrap",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Sync Without Source
        </span>
      </button>
    </div>
  </div>
  <div
    className="c6"
  />
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined c3 MuiButton-outlinedPrimary MuiButton-disableElevation"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Suspend
    </span>
  </button>
</div>
`;

exports[`SyncActions snapshots non-suspended 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 svg {
  height: 16px;
  width: 16px;
}

.c8 svg path.path-fill,
.c8 svg line.path-fill,
.c8 svg polygon.path-fill,
.c8 svg rect.path-fill,
.c8 svg circle.path-fill,
.c8 svg polyline.path-fill {
  fill: !important;
  -webkit-transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c8 svg path.stroke-fill,
.c8 svg line.stroke-fill,
.c8 svg polygon.stroke-fill,
.c8 svg rect.stroke-fill,
.c8 svg circle.stroke-fill,
.c8 svg polyline.stroke-fill {
  stroke: !important;
  -webkit-transition: stroke 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: stroke 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c8 svg rect.rect-height {
  height: 16px;
  width: 16px;
}

.c8.downward {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c8.upward {
  -webkit-transform: initial;
  -ms-transform: initial;
  transform: initial;
}

.c8 img {
  width: 16px;
}

.c10 {
  padding: 8px;
}

.c3.MuiButton-root {
  height: 32px;
  font-size: 12px;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  line-height: 1;
  border-radius: 2px;
  font-weight: 600;
}

.c3.MuiButton-outlined {
  padding: 8px 12px;
}

.c5.MuiButton-root {
  border-radius: 50%;
  min-width: 38px;
  height: 38px;
  padding: 0;
}

.c5.MuiButton-text {
  padding: 0;
}

.c6.MuiButton-outlined {
  border-radius: 0 2px 2px 0;
}

.c6.MuiButton-outlinedPrimary {
  border-color: #d8d8d8;
}

.c6.MuiButton-root {
  min-width: 0;
  height: 32px;
  padding: 8px 4px;
}

.c6.MuiButton-text {
  padding: 0;
}

.c4.MuiButton-outlined {
  margin-right: 0;
  border-radius: 2px 0 0 2px;
  border-right: none;
}

.c4.MuiButton-outlined.Mui-disabled {
  border-right: none;
}

.c9 {
  position: absolute;
  overflow: hidden;
  background: #fff;
  height: 0px;
  -webkit-transition-property: height;
  transition-property: height;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  z-index: 1;
}

.c1 {
  width: 50%;
}

<div
  className="c0 c1"
>
  <div
    className="SyncButton"
    style={
      Object {
        "display": "inline-block",
        "position": "relative",
      }
    }
  >
    <div
      className="c2"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined c3 c4 MuiButton-outlinedPrimary MuiButton-disableElevation"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Sync
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined c5 c6 MuiButton-outlinedPrimary MuiButton-disableElevation"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <div
            className="c7 c8"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </span>
      </button>
    </div>
    <div
      className="c2 c9"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined c3 MuiButton-outlinedPrimary MuiButton-disableElevation"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "whiteSpace": "nowrap",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Sync Without Source
        </span>
      </button>
    </div>
  </div>
  <div
    className="c10"
  />
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined c3 MuiButton-outlinedPrimary MuiButton-disableElevation"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Suspend
    </span>
  </button>
</div>
`;

exports[`SyncActions snapshots suspended 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 svg {
  height: 16px;
  width: 16px;
}

.c8 svg path.path-fill,
.c8 svg line.path-fill,
.c8 svg polygon.path-fill,
.c8 svg rect.path-fill,
.c8 svg circle.path-fill,
.c8 svg polyline.path-fill {
  fill: !important;
  -webkit-transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c8 svg path.stroke-fill,
.c8 svg line.stroke-fill,
.c8 svg polygon.stroke-fill,
.c8 svg rect.stroke-fill,
.c8 svg circle.stroke-fill,
.c8 svg polyline.stroke-fill {
  stroke: !important;
  -webkit-transition: stroke 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: stroke 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c8 svg rect.rect-height {
  height: 16px;
  width: 16px;
}

.c8.downward {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c8.upward {
  -webkit-transform: initial;
  -ms-transform: initial;
  transform: initial;
}

.c8 img {
  width: 16px;
}

.c10 {
  padding: 8px;
}

.c3.MuiButton-root {
  height: 32px;
  font-size: 12px;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  line-height: 1;
  border-radius: 2px;
  font-weight: 600;
}

.c3.MuiButton-outlined {
  padding: 8px 12px;
}

.c5.MuiButton-root {
  border-radius: 50%;
  min-width: 38px;
  height: 38px;
  padding: 0;
}

.c5.MuiButton-text {
  padding: 0;
}

.c6.MuiButton-outlined {
  border-radius: 0 2px 2px 0;
}

.c6.MuiButton-outlinedPrimary {
  border-color: #d8d8d8;
}

.c6.MuiButton-root {
  min-width: 0;
  height: 32px;
  padding: 8px 4px;
}

.c6.MuiButton-text {
  padding: 0;
}

.c4.MuiButton-outlined {
  margin-right: 0;
  border-radius: 2px 0 0 2px;
  border-right: none;
}

.c4.MuiButton-outlined.Mui-disabled {
  border-right: none;
}

.c9 {
  position: absolute;
  overflow: hidden;
  background: #fff;
  height: 0px;
  -webkit-transition-property: height;
  transition-property: height;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  z-index: 1;
}

.c1 {
  width: 50%;
}

<div
  className="c0 c1"
>
  <div
    className="SyncButton"
    style={
      Object {
        "display": "inline-block",
        "position": "relative",
      }
    }
  >
    <div
      className="c2"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined c3 c4 MuiButton-outlinedPrimary MuiButton-disableElevation Mui-disabled Mui-disabled"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Sync
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined c5 c6 MuiButton-outlinedPrimary MuiButton-disableElevation Mui-disabled Mui-disabled"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <div
            className="c7 c8"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </span>
      </button>
    </div>
    <div
      className="c2 c9"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined c3 MuiButton-outlinedPrimary MuiButton-disableElevation"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "whiteSpace": "nowrap",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Sync Without Source
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <div
    className="c10"
  />
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined c3 MuiButton-outlinedPrimary MuiButton-disableElevation"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Resume
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</div>
`;
